tests/ConstantTest.php

```diff
@@ -314,6 +314,51 @@ class Foo {
                 ',
                 'assertions' => ['$foo' => 'string'],
             ],
+            'dynamicClassConstFetch' => [
+                '<?php
+                    class Foo
+                    {
+                        public const BAR = "bar";
+                    }
+
+                    $foo = new Foo();
+                    $_trace = $foo::BAR;',
+                'assertions' => ['$_trace===' => '"bar"'],
+            ],
+            'unsafeInferenceClassConstFetch' => [
+                '<?php
+                    class Foo
+                    {
+                        public const BAR = "bar";
+                    }
+
+                    /** @var Foo $foo */
+                    $foo = new stdClass();
+                    $_trace = $foo::BAR;',
+                'assertions' => ['$_trace' => 'mixed'],
+            ],
+            'unsafeFinalInferenceClassConstFetch' => [
```

Comment:
Is it *un*safe?

Reply:
What I meant was it was unsafe on the definite_class side of things but final meant you could still make the inferrence, but you're right, it was not clear.